Some Persistence tests are failing with `Stubs process failed with exit code: 1`

From Persistence Platform TCK run with `mvn clean install -Djakarta.tck.arquillian.version=11.0.10  -Dtck.test.persistence.version=11.0.1-SNAPSHOT -f persistence-platform-tck-run/platform_pom.xml -Pee,staging   2>&1 | tee ee16.log`:

> INFO: Running appClient for: createEntityManagerSynchronizationTypeMapTest, monitor: DeploymentMonitor{name='jpa_ee_entityManager_appmanaged_vehicle', status=NEEDS_STUBS}
> May 21, 2025 9:02:54 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor invoke
> INFO: Getting deployment stubs for: jpa_ee_entityManager_vehicles_client.jar
> May 21, 2025 9:02:54 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Getting deployment stubs, deployment: jpa_ee_entityManager_appmanaged_vehicle
> May 21, 2025 9:02:54 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Running: /usr/lib/jvm/temurin-17-jdk/bin/java -jar /home/smarlow/tck/platformtck/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/target/glassfish8/glassfish/modules/admin-cli.jar get-client-stubs --appName jpa_ee_entityManager_appmanaged_vehicle target
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Stubs process exited with: 1
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientProtocol getExecutor
> INFO: getExecutor() called, config=tck.arquillian.protocol.appclient.AppClientProtocolConfiguration@68b4e7b5
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor invoke
> INFO: Running appClient for: createEntityManagerSynchronizationTypeTest, monitor: DeploymentMonitor{name='jpa_ee_entityManager_appmanaged_vehicle', status=NEEDS_STUBS}
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor invoke
> INFO: Getting deployment stubs for: jpa_ee_entityManager_vehicles_client.jar
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Getting deployment stubs, deployment: jpa_ee_entityManager_appmanaged_vehicle
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Running: /usr/lib/jvm/temurin-17-jdk/bin/java -jar /home/smarlow/tck/platformtck/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/target/glassfish8/glassfish/modules/admin-cli.jar get-client-stubs --appName jpa_ee_entityManager_appmanaged_vehicle target
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.AppClientMethodExecutor getDeploymentStubs
> INFO: Stubs process exited with: 1
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.DeploymentMonitorProvider onDeployment
> INFO: Deployment event observed: org.jboss.arquillian.container.spi.client.deployment.Deployment@47ffdbba, currentMonitor: DeploymentMonitor{name='jpa_ee_entityManager_appmanaged_vehicle', status=NEEDS_STUBS}
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.DeploymentMonitor dispose
> INFO: jpa_ee_entityManager_appmanaged_vehicle, disposed
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.DeploymentMonitor <init>
> INFO: jpa_ee_entityManager_appmanaged_vehicle, deployed
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.DeploymentMonitorProvider onUndeployment
> INFO: Undeployment event observed: org.jboss.arquillian.container.spi.client.deployment.Deployment@47ffdbba, currentMonitor: DeploymentMonitor{name='jpa_ee_entityManager_appmanaged_vehicle', status=NEEDS_STUBS}
> May 21, 2025 9:02:55 AM tck.arquillian.protocol.appclient.DeploymentMonitor undeploy
> INFO: jpa_ee_entityManager_appmanaged_vehicle, undeployed
> [ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 36.11 s <<< FAILURE! -- in ee.jakarta.tck.persistence.ee.entityManager.ClientAppmanagedTest
> [ERROR] ee.jakarta.tck.persistence.ee.entityManager.ClientAppmanagedTest.createEntityManagerSynchronizationTypeMapTest -- Time elapsed: 0.542 s <<< ERROR!
> java.lang.RuntimeException: Stubs process failed with exit code: 1
> 	at tck.arquillian.protocol.appclient.AppClientMethodExecutor.getDeploymentStubs(AppClientMethodExecutor.java:171)
> 	at tck.arquillian.protocol.appclient.AppClientMethodExecutor.invoke(AppClientMethodExecutor.java:96)
> 	at org.jboss.arquillian.container.test.impl.execution.RemoteTestExecuter.execute(RemoteTestExecuter.java:103)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)
> 	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
> 	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:52)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
> 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
> 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)
> 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
> 	at org.jboss.arquillian.junit5.ArquillianExtension.interceptInvocation(ArquillianExtension.java:167)
> 	at org.jboss.arquillian.junit5.ArquillianExtension.interceptTestMethod(ArquillianExtension.java:107)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
> 
> [ERROR] ee.jakarta.tck.persistence.ee.entityManager.ClientAppmanagedTest.createEntityManagerSynchronizationTypeTest -- Time elapsed: 0.451 s <<< ERROR!
> java.lang.RuntimeException: Stubs process failed with exit code: 1
> 	at tck.arquillian.protocol.appclient.AppClientMethodExecutor.getDeploymentStubs(AppClientMethodExecutor.java:171)
> 	at tck.arquillian.protocol.appclient.AppClientMethodExecutor.invoke(AppClientMethodExecutor.java:96)
> 	at org.jboss.arquillian.container.test.impl.execution.RemoteTestExecuter.execute(RemoteTestExecuter.java:103)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:106)
> 	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
> 	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:52)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
> 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
> 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
> 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
> 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
> 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:134)
> 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
> 	at org.jboss.arquillian.junit5.ArquillianExtension.interceptInvocation(ArquillianExtension.java:167)
> 	at org.jboss.arquillian.junit5.ArquillianExtension.interceptTestMethod(ArquillianExtension.java:107)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
> 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
> 
>

--- a/bom/pom.xml
+++ b/bom/pom.xml
@@ -41,7 +41,7 @@
     <jakarta.jakartaee-bom.version>11.0.0</jakarta.jakartaee-bom.version>
     <!-- Jakarta TCK Tools version from the independently released tools submodule -->
     <!-- The Arquillian protocol artifacts -->
-    <jakarta.tck.arquillian.version>11.0.10</jakarta.tck.arquillian.version>
+    <jakarta.tck.arquillian.version>11.0.11</jakarta.tck.arquillian.version>
     <!-- The released version of tools/common -->
     <jakarta.tck.common.version>11.0.8</jakarta.tck.common.version>
     <!-- The released version of tools/signaturetest -->

--- a/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/pom.xml
+++ b/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/pom.xml
@@ -42,7 +42,7 @@
         <glassfish.version>8.0.0-JDK17-M10</glassfish.version>
         <jakarta.platform.version>11.0.0-RC1</jakarta.platform.version>
 
-        <jakarta.tck.arquillian.version>11.0.10</jakarta.tck.arquillian.version>
+        <jakarta.tck.arquillian.version>11.0.11</jakarta.tck.arquillian.version>
         <jakarta.tck.common.version>11.0.8</jakarta.tck.common.version>
         <tck.test.persistence.version>11.0.1-M5</tck.test.persistence.version>
         
@@ -354,9 +354,12 @@
                         </goals>
                         <configuration>
                             <includes>
+                                <!--
+                                <include>ee.jakarta.tck.persistence.ee.entityManager.ClientPmservletTest</include>
+                                -->
                                 <include>ee/jakarta/tck/persistence/**/*Test.java</include>
                             </includes>
-                            <groups>${includeGroups}</groups>
+                            <groups>${includeGroups}&amp;tck-javatest</groups>
                             
                             <skipITs>${javatest.skip}</skipITs>
 
@@ -438,12 +441,14 @@
                                 <configuration>
                                     <includes>
                                         <!--
+                                        <include>ee.jakarta.tck.persistence.ee.entityManager.ClientAppmanagedTest</include>
                                         <include>ee.jakarta.tck.persistence.ee.packaging.appclient.annotation.ClientTest</include>
+                                        <include>ee.jakarta.tck.persistence.core.EntityGraph.ClientAppmanagedTest</include>
+                                        <include>ee.jakarta.tck.persistence.core.basic.ClientStateful3Test</include>
                                         -->
                                         <include>ee/jakarta/tck/persistence/**/*Test.java</include>
                                     </includes>
-                                    <!-- Should this not just be tck-appclient ? -->
-                                    <groups>${includeGroups}</groups>
+                                    <groups>${includeGroups}&amp;tck-appclient</groups>
                                     <excludedGroups>web</excludedGroups>
                                     
                                     <systemPropertyVariables>

--- a/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/src/test/java/test/jul/BetterFormatter.java
+++ b/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/src/test/java/test/jul/BetterFormatter.java
@@ -1,0 +1,45 @@
+package test.jul;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.logging.Formatter;
+import java.util.logging.LogRecord;
+
+public class BetterFormatter extends Formatter {
+    // date(1), source(2), logger(3), level(4), message(5), thrown(6)
+    private static final String CUSTOM_FORMAT = "[%4$s] %1$tF %1$tT %3$s : %5$s%6$s%n";
+    private static final int SOURCE_LENGTH = 20;
+
+    public BetterFormatter() {
+        System.out.println("BetterFormatter created");
+    }
+    @Override
+    public String format(LogRecord record) {
+        ZonedDateTime zdt = ZonedDateTime.ofInstant(
+                record.getInstant(), ZoneId.systemDefault());
+        String loggerName = record.getLoggerName();
+        // Get the last 12 characters of the source
+        if (loggerName != null && loggerName.length() > SOURCE_LENGTH) {
+            loggerName = loggerName.substring(loggerName.length() - SOURCE_LENGTH);
+        }
+        String message = formatMessage(record);
+        String throwable = "";
+        if (record.getThrown() != null) {
+            StringWriter sw = new StringWriter();
+            PrintWriter pw = new PrintWriter(sw);
+            pw.println();
+            record.getThrown().printStackTrace(pw);
+            pw.close();
+            throwable = sw.toString();
+        }
+        return String.format(CUSTOM_FORMAT,
+                             zdt,
+                             "",
+                             loggerName,
+                             record.getLevel().getName(),
+                             message,
+                             throwable);
+    }
+}

--- a/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/src/test/resources/logging.properties
+++ b/glassfish-runner/persistence-platform-tck/persistence-platform-tck-run/src/test/resources/logging.properties
@@ -1,0 +1,14 @@
+handlers=java.util.logging.FileHandler,java.util.logging.ConsoleHandler
+
+tck.jakarta.platform.ant.level = FINER
+
+java.util.logging.FileHandler.pattern = tck-run.log
+java.util.logging.FileHandler.level = FINEST
+#java.util.logging.FileHandler.formatter = tck.conversion.log.TestLogFormatter
+java.util.logging.FileHandler.formatter = test.jul.BetterFormatter
+# date(1), source(2), logger(3), level(4), message(5), thrown(6)
+#java.util.logging.SimpleFormatter.format = [%1$tH:%tM:%1$tS.%1$tL] %2$-.12s/%4$s %5$s %n
+
+# Limit the message that are printed on the console to INFO and above.
+java.util.logging.ConsoleHandler.level = INFO
+java.util.logging.ConsoleHandler.formatter = test.jul.BetterFormatter

--- a/pom.xml
+++ b/pom.xml
@@ -95,7 +95,7 @@
         <jakarta.jakartaee-bom.version>11.0.0</jakarta.jakartaee-bom.version>
         <!-- Jakarta TCK Tools version from the independently released tools submodule -->
         <!-- The Arquillian protocol artifacts -->
-        <jakarta.tck.arquillian.version>11.0.10</jakarta.tck.arquillian.version>
+        <jakarta.tck.arquillian.version>11.0.11</jakarta.tck.arquillian.version>
         <!-- The released version of tools/common -->
         <jakarta.tck.common.version>11.0.8</jakarta.tck.common.version>
         <!-- The released version of tools/signaturetest -->

